master/buildbot/data/resultspec.py

```diff
@@ -324,20 +348,39 @@ def includeFields(d):
             if total is None:
                 total = len(data)
 
-            # precompute the ordering functions and sort
             if self.order:
-                order = [(lambda a, b, k=k[1:]: nonecmp(b[k], a[k]))
-                         if k[0] == '-' else
-                         (lambda a, b, k=k: nonecmp(a[k], b[k]))
-                         for k in self.order]
-
-                def cmpFunc(a, b):
-                    for f in order:
-                        c = f(a, b)
-                        if c:
-                            return c
-                    return 0
-                data.sort(cmp=cmpFunc)
+                def keyFunc(elem, order=self.order,
+                            noneComparator=NoneComparator()):
```

Comment:
I am not sure why you pass NoneComparator in argument. I think it is a relique from a past design, we should simplify it.